### PR TITLE
Add support for Grafana library elements

### PR DESCRIPTION
**Pull request by amotl**
## About
This patch adds support for Grafana's Library Elements, by @dameiss-tibco. Thanks!

## References
- https://grafana.com/docs/grafana/latest/developers/http_api/library_element/
- https://grafana.com/docs/grafana/latest/dashboards/build-dashboards/manage-library-panels/

### CODE REVIEW

**Comment by amotl** [comment]
Dear @Ousret,

could you have a look what might be going on here?

```python
======================================================================
ERROR: test_create_library_element_already_exists (test.elements.test_libraryelement.LibraryElementTestCase.test_create_library_element_already_exists)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/site-packages/requests_mock/mocker.py", line 317, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/grafana-client/grafana-client/test/elements/test_libraryelement.py", line 920, in test_create_library_element_already_exists
    self.grafana.libraryelement.create_library_element(
  File "/home/runner/work/grafana-client/grafana-client/grafana_client/elements/libraryelement.py", line 90, in create_library_element
    return self.client.POST(create_element_path, json=json)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/grafana-client/grafana-client/grafana_client/client.py", line 198, in __request_runner
    r = self.s.request(
        ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/site-packages/niquests/sessions.py", line 539, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/site-packages/requests_mock/mocker.py", line 185, in _fake_send
    return _original_send(session, request, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/site-packages/niquests/sessions.py", line 1189, in send
    r._promise.set_parameter("niquests_is_stream", stream)
    ^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/site-packages/niquests/models.py", line 1028, in __getattribute__
    return super().__getattribute__(item)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Response' object has no attribute '_promise'
```
-- https://github.com/panodata/grafana-client/actions/runs/8480351660/job/23235924117?pr=168#step:7:509

File: test/elements/test_libraryelement.py [hunk 1]
```diff
@@ -0,0 +1,1123 @@
...
+
+        m.get(
+            "http://localhost/api/health",
+            json=LibraryElementTestCase.HealthResponsePost8_2,
+        )
+        m.add_matcher(custom_matcher)
+        with self.assertRaises(GrafanaClientError) as ex:
+            self.grafana.libraryelement.get_library_element(LibraryElementTestCase.MissingPanelUID)
+        self.assertEqual(404, ex.exception.status_code)
+        self.assertEqual("Client Error 404: library element could not be found", ex.exception.message)
```
Comment (amotl) [inline test/elements/test_libraryelement.py:647]: I guess the reason for the hiccup is coming from this slightly different way to establish a request mock. I already tried to use `resp = niquests.Response()` here, but it still shows the same error.

```python
AttributeError: 'Response' object has no attribute '_promise'
```

Comment (Ousret) [reply test/elements/test_libraryelement.py:647]: Immediately, yes, the mocking is the issue here.
The "raw" attribute must be set within the response, even if it's a empty BytesIO.

I am going to look deeper into it as soon as I can.

Comment (amotl) [reply test/elements/test_libraryelement.py]: I've replaced it by using the regular way of mocking. Might be an edge case to consider in order to improve interaction between niquests and requests_mock.

Comment (Ousret) [reply test/elements/test_libraryelement.py]: OK. I've noted it on my side.

Comment (Ousret) [reply test/elements/test_libraryelement.py]: The CI failure is unrelated to us. httpbin.org flakyness.

Comment (amotl) [reply test/elements/test_libraryelement.py]: Fixed with e1c4358.
